wallet/bench/Main.hs

```diff
@@ -0,0 +1,31 @@
+module Main
+  ( main
+  ) where
+
+import           Universum
+
+import           Control.Concurrent.Async       (async, wait)
+import           System.IO                      (hSetEncoding, stdout, utf8)
+import           Data.Maybe                     (maybe)
+
+import           Bench.Pos.Wallet.Config        (extractConfigFor, getBenchConfig)
+import           Bench.Pos.Wallet.Run           (runBench)
+import           Bench.Pos.Wallet.Types         (BenchEndpoint (..))
+import           Client.Pos.Wallet.Web.Endpoint (getHistoryIO, getWalletsIO, newPaymentIO)
+
+-- | Use `stack bench cardano-sl-wallet` to run this benchmark.
+-- It's a client, so we assume that the node (with Wallet Web API enabled)
+-- is already running.
+main :: IO ()
+main = do
+    config <- getBenchConfig
+    hSetEncoding stdout utf8
+    putText "Wallet benchmarking begin..."
+    -- Only benchmark with config will be launched.
+    b1 <- async $ maybe (return ()) (runBench getHistoryIO) (extractConfigFor GetHistoryBench config)
+    b2 <- async $ maybe (return ()) (runBench getWalletsIO) (extractConfigFor GetWalletsBench config)
+    b3 <- async $ maybe (return ()) (runBench newPaymentIO) (extractConfigFor NewPaymentBench config)
```

Comment:
Are we sure we want to run all the benchmarks concurrently? Perhaps it would be better sequentially, so that we get less variance out of them.

Reply:
> Are we sure we want to run all the benchmarks concurrently? 

AFAIU, yes. Because in the real life all requests always occurs concurrently... :-)

Reply:
Yes, I agree with that, but how do you get consistent measurements out of a benchmark if the starting environment is not stable? In order to get meaningful data out of an experiment (a benchmark in this case) you have to lock in place variance as much as you can, otherwise it's not a bench anymore but a stress test 😉  This is just my view and I'm definitely not the foremost expert, but I'm sure people like Neil can advise here 😀